test/framework/machinedeployment_helpers.go

```diff
@@ -289,3 +290,62 @@ func machineDeploymentOptions(deployment clusterv1.MachineDeployment) []client.L
 		client.MatchingLabels(deployment.Spec.Selector.MatchLabels),
 	}
 }
+
+// ScaleAndWaitMachineDeploymentInput is the input for ScaleAndWaitMachineDeployment.
+type ScaleAndWaitMachineDeploymentInput struct {
+	ClusterProxy              ClusterProxy
+	Cluster                   *clusterv1.Cluster
+	MachineDeployment         *clusterv1.MachineDeployment
+	Replicas                  int32
+	WaitForMachineDeployments []interface{}
+}
+
+// ScaleAndWaitMachineDeployment scales MachineDeployment and waits until all machines have node ref and equal to Replicas.
+func ScaleAndWaitMachineDeployment(ctx context.Context, input ScaleAndWaitMachineDeploymentInput) {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for ScaleAndWaitMachineDeployment")
+	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling ScaleAndWaitMachineDeployment")
+	Expect(input.Replicas).ToNot(BeNil(), "Invalid argument. input.Replicas can't be empty when calling ScaleAndWaitMachineDeployment")
```

Comment:
```suggestion
	Expect(input.Replicas).ToNot(BeEmpty(), "Invalid argument. input.Replicas can't be empty when calling ScaleAndWaitMachineDeployment")
```
I'm not sure of the behavior of BeNil with integers...

Reply:
You are right, we should not even add Empty check as we would want to support scale down to 0. Removed it.

Reply:
Might be test if replicas ≥ 0

Reply:
If replicas is not provided, it comes as 0 so no need to add this check.